scripts/commands/additem.lua

```diff
@@ -6,35 +6,55 @@
 cmdprops =
 {
     permission = 1,
-    parameters = "iiiiiiiiiii"
+    parameters = "siiiiiiiiii"
 }
 
 function error(player, msg)
     player:PrintToPlayer(msg)
     player:PrintToPlayer("!additem <itemId> {quantity} {aug1} {v1} {aug2} {v2} {aug3} {v3} {aug4} {v4} {trial}")
 end
 
-function onTrigger(player, itemId, quantity, aug0, aug0val, aug1, aug1val, aug2, aug2val, aug3, aug3val, trialId)
+function onTrigger(player, item, quantity, aug0, aug0val, aug1, aug1val, aug2, aug2val, aug3, aug3val, trialId)
     -- Load needed text ids for players current zone..
     local ID = zones[player:getZoneID()]
+    local itemToGet = 0
 
-    -- validate itemId
-    if (itemId == nil or tonumber(itemId) == nil or tonumber(itemId) == 0) then
-        error(player, "Invalid itemId.")
+    -- validate item
+    if item == nil then
+        -- No Item Provided
+        error(player, "No Item ID given.")
+        return
+    elseif tonumber(item) == nil and item ~= nil then
+        -- Item was provided, but was not a number.  Try text lookup.
+        local retItem = GetItemIDByName(tostring(item))
+        if retItem > 0 then
+            itemToGet = retItem
+        else
+            error(player, string.format("Item %s not found in database.", item))
+            return
+        end
+    else
+        -- Number was provided, so just use it
+        itemToGet = tonumber(item)
+    end
+
+    -- At this point, if there's no item found, exit out of the function
+    if itemToGet == 0 then
+        error(player, "Item not found.")
         return
     end
 
     -- Ensure the GM has room to obtain the item...
-    if (player:getFreeSlotsCount() == 0) then
-        player:messageSpecial( ID.text.ITEM_CANNOT_BE_OBTAINED, itemId )
+    if player:getFreeSlotsCount() == 0 then
+        player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED, item)
         return
     end
 
     -- Give the GM the item...
-    player:addItem( itemId, quantity, aug0, aug0val, aug1, aug1val, aug2, aug2val, aug3, aug3val, trialId )
+    player:addItem(itemToGet, quantity, aug0, aug0val, aug1, aug1val, aug2, aug2val, aug3, aug3val, trialId)
     if quantity and quantity > 1 then
-        player:messageSpecial( ID.text.ITEM_OBTAINED + 9, itemId , quantity )
+        player:messageSpecial(ID.text.ITEM_OBTAINED + 9, itemToGet, quantity)
     else
-        player:messageSpecial( ID.text.ITEM_OBTAINED, itemId )
+        player:messageSpecial(ID.text.ITEM_OBTAINED, itemToGet)
     end
-end
+end
```

Comment:
Style: Add a newline at the end, pls

Reply:
New line added after end

Reply:
If you have WSL setup or develop on linux, you can run the CI locally:
`tools/ci/lua.sh`

you can ignore the `cyclomatic complexity` errors on files you haven't edited

Reply:
I don't have a linux dev env, unfortunately.  If it's a requirement, I can add one in though.

Reply:
nope, it's just to help you fix up CI issues before you push to make things easier on you

Reply:
Awesome, I'll look into it.  Thanks!